data2bids.m

```diff
@@ -1323,10 +1334,11 @@
 
 %% need_motion_json
 if need_motion_json
-  motion_json.SamplingFrequency = hdr.Fs;
-  motion_json.StartTime = nan;
-  motion_json.Columns = hdr.label;
-  motion_json.RecordingDuration = (hdr.nSamples*hdr.nTrials)/hdr.Fs;
+    
+  motion_json.SamplingFrequency     = hdr.Fs;
+  motion_json.StartTime             = nan;
+  motion_json.MotionChannelCount    = hdr.nChans; 
+  motion_json.RecordingDuration     = hdr.nSamples/hdr.Fs; 
```

Comment:
it should in general be `hdr.nSamples*hdr.nTrials` rather than `hdr.nSamples` to deal with data that is represented in segments. That might not happen for common motion formats, but is e.g. the case for EDF (which stores data in 1-second chunks) or with epoch FIF files.